### PR TITLE
os/fs: Fix SAM issues in SmartFS finddirdntry Function

### PR DESCRIPTION
**Pull request by KimDongEon**
Cyclomatic Complexities of file system functions are reduced
by defining parts of the function as indivisual functions.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [25a380be8f5e917d5bc248e52eee6c4fbe0d4d8a] - Code Rule Check Result:
os/fs/smartfs/smartfs_utils.c:1035: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/fs/smartfs/smartfs_utils.c:1042: ERROR: [BRC_M_SMT] that open brace { should be on the previous line

**Comment by seinfra** [comment]
Target : [25a380be8f5e917d5bc248e52eee6c4fbe0d4d8a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0ec860cbf1cee8f8fc14bd5bb5e59d58905e69b3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0ec860cbf1cee8f8fc14bd5bb5e59d58905e69b3] - Code Rule Check OK.

File: os/fs/smartfs/smartfs_utils.c [hunk 1]
```diff
@@ -257,8 +256,8 @@ uint16_t smartfs_rdle16(FAR const void *val)
 
 void smartfs_wrle16(FAR void *dest, uint16_t val)
 {
-	((FAR uint8_t *)dest)[0] = val & 0xff;	/* Little endian means LS byte first in byte stream */
-	((FAR uint8_t *)dest)[1] = val >> 8;
+	((FAR uint8_t *) dest)[0] = val & 0xff;	/* Little endian means LS byte first in byte stream */
```
Comment (jeongchanKim) [inline os/fs/smartfs/smartfs_utils.c]: I think the original is correct for our coding rule.

**Pull request closed without merging**